### PR TITLE
fix: register withdraw emissions to legacy amino

### PR DESCRIPTION
**Pull request by kingpinXD**
# Description

Closes https://github.com/zeta-chain/node/issues/3116

# How Has This Been Tested?

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions

## Summary by CodeRabbit

- **New Features**
	- Added the ability to whitelist SPL tokens on Solana.
- **Improvements**
	- Enhanced build reproducibility with stable checksums during the release process.
- **New Tests**
	- Introduced tests for concurrent withdrawals, deposits, and reverts.
- **Bug Fixes**
	- Improved compatibility by registering messages for the emissions module to the legacy amino codec.
	- Replaced public DHT with a private gossip peer discovery mechanism for enhanced security and performance.

### CODE REVIEW

File: changelog.md [hunk 1]
```diff
@@ -75,6 +75,7 @@
 * [3018](https://github.com/zeta-chain/node/pull/3018) - support `DepositAndCall` and `WithdrawAndCall` with empty payload
 * [3030](https://github.com/zeta-chain/node/pull/3030) - Avoid storing invalid Solana gateway address in the `SetGatewayAddress`
 * [3047](https://github.com/zeta-chain/node/pull/3047) - wrong block hash in subscribe new heads
+* [3166](https://github.com/zeta-chain/node/pull/3117) - register withdraw emissions to legacy amino
```
Comment (lumtis) [inline changelog.md:78]: Wrong location

File: x/emissions/types/codec.go [hunk 2]
```diff
@@ -3,13 +3,18 @@ package types
 import (
 	"github.com/cosmos/cosmos-sdk/codec"
 	cdctypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/types/msgservice"
 )
 
-func RegisterCodec(_ *codec.LegacyAmino) {
+func RegisterCodec(cdc *codec.LegacyAmino) {
+	cdc.RegisterConcrete(&MsgWithdrawEmission{}, "emissions/WithdrawEmission", nil)
```
Comment (lumtis) [inline x/emissions/types/codec.go:11]: Let's add the other message as well https://github.com/zeta-chain/node/issues/3116#issuecomment-2460214181

**Review by lumtis — CHANGES_REQUESTED** [review]

File: x/emissions/types/codec.go (hunk 2 above)
Comment (kingpinXD) [reply x/emissions/types/codec.go:11]: Already addressed in a subsequent commit

**Comment by lumtis** [comment]
@kingpinXD  Any update on this one? Still draft?

**Comment by kingpinXD** [comment]
> @kingpinXD  Any update on this one? Still draft?

I think the original issue posted was because of the wrong format of the proposal ,
We still need to add this change but I am waiting for the trying out the execution on mainnet to verify if we run into any other errors 
This pr is ready though, I can open/merge it if you want

**Comment by lumtis** [comment]
> > @kingpinXD  Any update on this one? Still draft?
> 
> I think the original issue posted was because of the wrong format of the proposal , We still need to add this change but I am waiting for the trying out the execution on mainnet to verify if we run into any other errors This pr is ready though, I can open/merge it if you want

I think we can look toward making this one open

**Review by lumtis — APPROVED** [review]

**Review by fbac — APPROVED** [review]

File: changelog.md [hunk 3]
```diff
@@ -20,6 +20,7 @@
 * [3137](https://github.com/zeta-chain/node/pull/3137) - remove chain.Chain from zetaclientd config
 
 ### Fixes
+* [3166](https://github.com/zeta-chain/node/pull/3117) - register messages for emissions module to legacy amino codec.
```
Comment (lumtis) [inline changelog.md:23]:
```suggestion
* [3117](https://github.com/zeta-chain/node/pull/3117) - register messages for emissions module to legacy amino codec.
```

**Pull request merged**